### PR TITLE
Feature: meta transactions

### PR DESCRIPTION
**Pull request by smokeynotes**
This introduces a new `meta` service with support for different meta transaction providers, the first of which is Biconomy.  The meta service, if supplied with appropriate configuration, will wrap the existing `ethersConnection.signer` object and automatically send all allowed functions through the configured meta transaction service, falling back gracefully to using the default signer.

A custom `estimateGas` function and `estimateGasCallback` handler has been introduced to allow a configured meta transaction provider to customize the `gasLimit`, which is currently needed for Biconomy since the default `estimateGas` function doesn't take into account the extra gas needed by the relay operations.  For now, the `gasLimit` for Biconomy will automatically be increased by `15%` to mitigate this issue.

### CODE REVIEW

File: src/meta/Meta.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                            let gasLimit = gas?.toNumber();
+                            let newGas = gasLimit * this.gasLimitMargin;
+                            return ethers.BigNumber.from(parseInt(newGas));
+                        }
+                        return gas;
+                    });
+                    this.logger.info(`Biconomy estimateGas patched.`);
+                } catch (error) {
+                    this.logger.info(`Biconomy failed to initialize.`);
+                    console.dir(error);
```
Comment (masha256) [inline src/meta/Meta.js:85]: @smokeynotes I've been keeping `console` messages out of the SDK. Can you just this.logger.warn that instead?

File: src/meta/Meta.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                    break;
+                }
+                if (ethersConnection.signer?.provider?.provider?.isBiconomy) {
+                    // already setup
+                    this.logger.info(`Biconomy already initalized.`);
+                    return;
+                }
+
+                try {
+                    this.biconomy = new Biconomy(ethersConnection.signer.provider.provider, {
```
Comment (masha256) [inline src/meta/Meta.js:67]: @smokeynotes does this work with WalletConnect?

**Review by masha256 — CHANGES_REQUESTED** [review]

File: src/meta/Meta.js (hunk 1 above)
Comment (smokeynotes) [reply src/meta/Meta.js:85]: Will do.

File: src/meta/Meta.js (hunk 2 above)
Comment (smokeynotes) [reply src/meta/Meta.js:67]: Haven't tested with WC yet.  Wanted to give you the chance to raise any concerns with this implementation before I do more extensive testing.

Comment (masha256) [reply src/meta/Meta.js:67]: @smokeynotes looks great, lets merge and see if WC works!

**Review by masha256 — APPROVED** [review]

**Pull request merged**